### ISSUE TITLE
Dynamo AutocastModeVariable bug: missing with context on the graph break instruction

### ISSUE THREAD

**Issue by yanboliang** [pytorch/pytorch#95837]
### 🐛 Describe the bug

This bug was found when i run Dynamo benchmarks after #95416. Since Dynamo benchmark AMP tests are using ```torch.cuda.amp.autocast``` which is not supported before #95416, so it falls back to eager mode and this bug was hidden.

Repro:
```
import torch
import torch._dynamo

def fn(x):
    torch._dynamo.graph_break()
    return x.sum() / x.numel()

class MyModule(torch.nn.Module):
    def forward(self, a, b):
        with torch.amp.autocast(device_type="cuda"):
            x = a + b
            y1 = fn(x)
        return y1

module = MyModule()
a = torch.rand((8, 8), dtype=torch.float16, device="cuda")
b = torch.rand((8, 8), dtype=torch.float16, device="cuda")
opt_m = torch._dynamo.optimize("eager")(module)
print(module(a, b).dtype)
print(opt_m(a, b).dtype)
```
Output
```
torch.float32
torch.float16
```

The root cause is exactly the same as https://github.com/pytorch/torchdynamo/issues/207, but for ```AutocastModeVariable```.

### Versions

N/A

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

**Comment by yanboliang** [comment]
#94758 potential can fix this.

**Comment by jon-chuang** [comment]
Indeed, https://github.com/pytorch/pytorch/pull/94758 is designed to fix exactly this. I'll try to devote a few cycles this week/weekend to the backlog I've accumulated for Pytorch.

**Comment by yanboliang** [comment]
Closing as this has been fixed by #94758
